src/parser/function.js

```diff
@@ -160,6 +160,12 @@ module.exports = {
    * ```
    */
   read_lexical_var: function () {
+    // a trailing separator is allowed in PHP8, which would put us right on the
+    // closing parenthesis. the ")" can be ignored here, the actually closing
+    // of the list is done in `read_lexical_vars`
+    if (this.version >= 800 && this.token === ")") {
+      return;
```

Comment:
Why return? I think it is wrong solution

Reply:
`read_list` moves past the dangling comma and runs this function. And the rest of this function expects a variable, but that will never come if there is a `)`, so we need to exit early when this is the case.

Reply:
I think we need put this in other place, before running this function, otherwise we have mixed logic

Reply:
I agree that this solution is not ideal.

The `read_list` function has no knowledge about the enclosing parenthesis, shall I create a specialized `read_list` for this purpose, like is done for the parameters of a function (`read_parameter_list`)?

Or do you have a different solution in mind?